src/schema/types/product.graphql

```diff
@@ -0,0 +1,12 @@
+type Product {
+  id: ID!
+  photoUrl: String!
+  name: String!
+  description: String
+  price: Float!
+  marketPrice: Float!
+}
+
+extend type Query {
+  products: [Product]!
```

Comment:
This should be `[Products!]!`.

Reply:
ummm ok, the product array can not be empty

Reply:
It actually can. The double bang requires the controller to return either an array or nothing. With only one (`[]!`) the controller could return an array of empty values in it: `[product1, product2, null, undefined, false]`.

Reply:
Aaah! Got it! Thanks @abelosorio